src/micro_http/src/common/headers.rs

```diff
@@ -199,6 +207,24 @@ impl Headers {
                             )),
                         },
                         Header::Server => Ok(()),
+                        Header::AcceptEncoding => {
+                            match Encoding::try_from(entry[1].trim().as_bytes()) {
```

Comment:
I think the match arms can be better expressed with `map` and `map_err`. This would also reduce the indentation level.